blockchain/indexers/utreexoproofindex.go

```diff
@@ -118,15 +118,19 @@ func (idx *UtreexoProofIndex) Create(dbTx database.Tx) error {
 // This is part of the Indexer interface.
 func (idx *UtreexoProofIndex) ConnectBlock(dbTx database.Tx, block *btcutil.Block,
 	stxos []blockchain.SpentTxOut) error {
-
+	pruneHeight := uint64(block.Height())
+	tip := uint64(idx.chain.BestSnapshot().Height)
 	// Don't include genesis blocks.
 	if block.Height() == 0 {
 		log.Tracef("UtreexoProofIndex.ConnectBlock: Asked to connect genesis"+
 			" block (height %d) Ignoring request and skipping block",
 			block.Height())
 		return nil
 	}
-
+	// Don't save proof if we are past the prune height.
+	if (tip - idx.chain.GetPruneTarget()) > pruneHeight {
```

Comment:
The `pruneTarget` isn't the height but it's the target size of bytes the database should be so this doesn't work.